modules/unsupported/geojson-core/src/test/java/org/geotools/filter/expression/JSONNodePropertyAccessorTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2022, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.expression;
+
+import static org.junit.Assert.assertEquals;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.DoubleNode;
+import com.fasterxml.jackson.databind.node.IntNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import java.net.URL;
+import java.util.Date;
+import org.geotools.data.geojson.GeoJSONReader;
+import org.geotools.data.geojson.GeoJSONReaderTest;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.test.TestData;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+
+public class JSONNodePropertyAccessorTest {
+
+    SimpleFeatureType type;
+    SimpleFeature feature;
+    PropertyAccessor accessor = SimpleFeaturePropertyAccessorFactory.ATTRIBUTE_ACCESS;
+
+    @Before
+    public void setUp() throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
+        SimpleFeatureTypeBuilder typeBuilder = new SimpleFeatureTypeBuilder();
+
+        typeBuilder.setName("test");
+        typeBuilder.setNamespaceURI("http://www.geotools.org/test");
+        typeBuilder.add("aString", String.class);
+        typeBuilder.add("aTextNode", TextNode.class);
+        typeBuilder.add("aThreeLevelObject", ObjectNode.class);
+        typeBuilder.add("aTwoLevelObject", ObjectNode.class);
+        typeBuilder.add("aTwoLevelObjectNoChildren", ObjectNode.class);
+        typeBuilder.add("aDateAsText", TextNode.class);
+        typeBuilder.add("aThreeLevelObjectWithDouble", ObjectNode.class);
+        typeBuilder.add("aArray", ArrayNode.class);
+        typeBuilder.add("aArrayWithObject", ArrayNode.class);
+
+        type = typeBuilder.buildFeatureType();
+
+        SimpleFeatureBuilder builder = new SimpleFeatureBuilder(type);
+
+        builder.add("testString");
+
+        builder.add(new TextNode("testTextNode"));
+
+        ObjectNode threeLevelObject = objectMapper.createObjectNode();
```

Comment:
Works, but it's a bit hard to imagine the resulting JSON... Wouldn't it have been easier to write a small JSON snippet, and have ObjectMapper parse it instead?

Reply:
Moved to json in a seperate file as a SimpleFeature for better readability.